src/utils/files.helper.ts

```diff
@@ -117,35 +117,42 @@ export function removeFile(options: string, files: string[]): void {
   }
 }
 
+const logger = new OrbitLogger('FILE-REPLACER');
 /**
  * Recursively replaces a specified string with another in all files within a directory.
  *
- * @param {string} dir - Directory to search for files.
- * @param {string} searchStr - The string to search for in the files.
- * @param {string} replaceStr - The string to replace the searchStr with.
+ * @param {object} options - Options for the replaceInFiles function.
+ * @param {string} options.dir - Directory to search for files.
+ * @param {string} options.searchStr - The string to search for in the files.
+ * @param {string} options.replaceStr - The string to replace the searchStr with.
  */
-export function replaceInFiles(
-  dir: string,
-  searchStr: string,
-  replaceStr: string,
-) {
-  const logger = new OrbitLogger('FILE-REPLACER');
+export function replaceInFiles({
+  dir,
+  searchStr,
+  replaceStr,
+}: {
+  dir: string;
+  searchStr: string;
+  replaceStr: string;
+}) {
+  if (searchStr === replaceStr) return;
 
   try {
+    // The sync fs functions are faster for this use case
```

Comment:
One of the things I did was ensure that `replaceInFiles` was fast despite a template having a large amount of files. The sync version of the function is 10x faster. The `if(content.match(searchStr))` line below helped in that.